app/Adapters/Task.php

```diff
@@ -28,6 +28,42 @@ public function process()
         $this->task->estimatedHours = sprintf('%.2f', $this->task->estimatedHours);
         $this->task->xp = sprintf('%.2f', $this->task->xp);
 
+        $taskStatus = $profilePerformance->perTask($this->task);
+
+        if (!empty($taskStatus)) {
+            $taskEstimatedSeconds = $mappedValues['estimatedHours'] * 60 * 60;
+            $taskDeliveredOnTime = $taskStatus[$this->task->owner]['workSeconds']
+            <= $taskEstimatedSeconds ? true : false;
+
+            // deadline in last 25% of the time of task
+            $lastQuarterOfTask = (25 / 100) * $taskEstimatedSeconds;
+
+            //generate task color status
+            if ($this->task->passed_qa === true && $taskDeliveredOnTime === true) {
+                $this->task->colorIndicator = 'Green';
+                return $this->task;
+            }
+
+            if ($this->task->passed_qa === true && $taskDeliveredOnTime === false) {
```

Comment:
This should be the last check, this way if you submit late, qa passes it's going to render `blue`